scripts/protobuf.go

```diff
@@ -148,9 +148,12 @@ func generate(dir string, dirs []string, files []string) error {
 }
 
 func appendCommonArguments(args []string, dir string, dirs []string, files []string) []string {
+	var includes []string
+	prependDirInclude := false
 	for _, otherdir := range dirs {
 		if otherdir == dir {
-			args = append(args, "-I=.")
+			// the include for the directory must be added first... see below.
+			prependDirInclude = true
```

Comment:
It looks that we could have a simpler logic (IMO), not having the _bool prependDirInclude_ and leave the `args ==append(args, "-I=.")` and only add the code which appends `includes` and at the end appends `includes` to `args`

Reply:
I haven't dug through the callers enough to understand if `dir` should always be included in the proto includes. If it can, then I think we can take your suggestion to unconditionally append `-I=.` first (and skip adding it a second time if `dir` is in `dirs`). My fix was maintaining existing semantics other than the ordering of the includes.